boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig

```diff
@@ -59,26 +59,53 @@ endif # BUILD_WITH_TFM
 # always restricted to the allocated non-secure SRAM partition.
 #
 # Workaround for not being able to have commas in macro arguments
+# Workaround for not being able to have commas in macro arguments
 DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+DT_CHOSEN_Z_SINGLE_IMAGE_CODE_PARTITION := zephyr,single-image-code-partition
+DT_CHOSEN_Z_CODE_S_PARTITION := zephyr,code-s-partition
+DT_CHOSEN_Z_SINGLE_IMAGE_CODE_S_PARTITION := zephyr,single-image-code-s-partition
 DT_CHOSEN_Z_SRAM_PARTITION := zephyr,sram-secure-partition
 
-if (BOARD_NRF5340PDK_NRF5340_CPUAPP || BOARD_NRF5340DK_NRF5340_CPUAPP) && TRUSTED_EXECUTION_SECURE
+if (BOARD_NRF5340PDK_NRF5340_CPUAPP || BOARD_NRF5340DK_NRF5340_CPUAPP)
+
+config HAS_DT_CODE_PARTITION
+	bool
+	default $(dt_chosen_enabled,$(DT_CHOSEN_Z_CODE_S_PARTITION)) if TRUSTED_EXECUTION_SECURE && BOOTLOADER_MCUBOOT
+	default $(dt_chosen_enabled,$(DT_CHOSEN_Z_SINGLE_IMAGE_CODE_S_PARTITION)) if TRUSTED_EXECUTION_SECURE
+	default $(dt_chosen_enabled,$(DT_CHOSEN_Z_CODE_PARTITION)) if TRUSTED_EXECUTION_SECURE && BOOTLOADER_MCUBOOT
+	default $(dt_chosen_enabled,$(DT_CHOSEN_Z_SINGLE_IMAGE_CODE_PARTITION)) if TRUSTED_EXECUTION_SECURE
```

Comment:
The number of cases that are influencing HAS_DT_CODE_PARTITION is starting to explode. Not sure about this either.

Reply:
This is a special case for TrustZone-enable devices that want to support both running standalone Secure-only execution environments and Secure/Non-Secure environments.